### PR TITLE
C353635 Verify that user can bulk edit permanent and temporary locati…

### PR DESCRIPTION
**Pull request by KarapetyanSofia**
[C353635](https://foliotest.testrail.io/index.php?/cases/view/353635)

[Allure Report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/5644/allure)

image

### CODE REVIEW

File: cypress/support/fragments/bulk-edit/bulk-edit-actions.js [hunk 1]
```diff
@@ -249,7 +252,16 @@ export default {
       SelectionOption(including(location)).click(),
     ]);
   },
-
+  selectLocation(location, rowIndex) {
```
Comment (nayimovag) [inline cypress/support/fragments/bulk-edit/bulk-edit-actions.js:255]: see this method `replacePermanentLocation`

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-permanent-temporary-location-at-once.cy.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      ServicePoints.createViaApi(testData.servicePoint);
+      testData.defaultLocation = Location.getDefaultLocation(testData.servicePoint.id);
+      Location.createViaApi(testData.defaultLocation);
+      cy.createTempUser([
+        Permissions.bulkEditEdit.gui,
+        Permissions.uiInventoryViewCreateEditDeleteItems.gui,
+      ]).then((userProperties) => {
+        user = userProperties;
+        InventoryInstances.createInstanceViaApi(item.instanceName, item.firstBarcode);
+        cy.getItems({
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-permanent-temporary-location-at-once.cy.js:45]: why do you need this?

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-permanent-temporary-location-at-once.cy.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        BulkEditActions.locationLookupExists();
+        BulkEditActions.clickLocationLookup(1);
+        BulkEditActions.verifyLocationLookupModal();
+        Locations.selectInstitution(testData.defaultLocation.institutionName);
+        BulkEditActions.locationLookupModalSaveAndClose();
+        // Click "Confirm changes" button
+        BulkEditActions.confirmChanges();
+        BulkEditActions.verifyAreYouSureForm(1, item.firstBarcode);
+        // Download preview
+        BulkEditActions.downloadPreview();
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-permanent-temporary-location-at-once.cy.js:111]: are you opening up these downloaded files and checking the content?

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-permanent-temporary-location-at-once.cy.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      cy.getAdminToken();
+      ServicePoints.createViaApi(testData.servicePoint);
+      testData.defaultLocation = Location.getDefaultLocation(testData.servicePoint.id);
+      Location.createViaApi(testData.defaultLocation);
+      cy.createTempUser([
+        Permissions.bulkEditEdit.gui,
+        Permissions.uiInventoryViewCreateEditDeleteItems.gui,
+      ]).then((userProperties) => {
+        user = userProperties;
+        InventoryInstances.createInstanceViaApi(item.instanceName, item.firstBarcode);
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-permanent-temporary-location-at-once.cy.js:44]: are you maiking sure this item has empty permanent and temporary locations as required in the TC?

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-permanent-temporary-location-at-once.cy.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+const changedRecordsFileName = `*-Changed-Records-${itemBarcodesFileName}`;
+const matchedRecordsFileName = `*Matched-Records-${itemBarcodesFileName}`;
+const previewFileName = `*-Updates-Preview-${itemBarcodesFileName}`;
+const testData = {
+  folioInstances: InventoryInstances.generateFolioInstances(),
+  servicePoint: ServicePoints.getDefaultServicePointWithPickUpLocation(),
+  requestsId: '',
+};
+
+describe('Bulk Edits', () => {
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-permanent-temporary-location-at-once.cy.js:32]: please follow this notation, as other tests in the folder
`describe('bulk-edit', () => {
  describe('in-app approach', () => {`

File: cypress/support/fragments/bulk-edit/bulk-edit-actions.js (hunk 1 above)
Comment (KarapetyanSofia) [reply cypress/support/fragments/bulk-edit/bulk-edit-actions.js:255]: I`ve seen that method, but in TC step by step I should do

**Review by manowarqqq — APPROVED** [review]

**Pull request merged**